tests/boot/grub_test_snapshot.pm

```diff
@@ -14,28 +14,39 @@ use testapi;
 
 sub run() {
     my $self = shift;
-
     assert_screen "grub2";
     # prevent grub2 timeout; 'esc' would be cleaner, but grub2-efi falls to the menu then
+
     send_key 'up';
     if (get_var("BOOT_TO_SNAPSHOT")) {
         send_key_until_needlematch("boot-menu-snapshot", 'down', 10, 5);
         send_key 'ret';
         # find out the before migration snapshot
-        send_key_until_needlematch("snap-before-update",    'down', 40, 5) if (get_var("UPGRADE"));
-        send_key_until_needlematch("snap-before-migration", 'down', 40, 5) if (get_var("MIGRATION_ROLLBACK"));
+        send_key 'end';
+        send_key_until_needlematch("snap-before-update",    'up', 40, 5) if (get_var("UPGRADE"));
+        send_key_until_needlematch("snap-before-migration", 'up', 40, 5) if (get_var("MIGRATION_ROLLBACK"));
         send_key "ret";
+        send_key 'up';    #stop grub
+
         # bsc#956046  check if we are in first menu-entry, or not
         if (check_screen("boot-menu-snapshot-bootmenu")) {
             record_soft_failure 'bsc#956046';
             send_key 'down', 1;
             save_screenshot;
         }
-        send_key 'ret';
+
+        send_key 'e';
+        save_screenshot;
+        send_key_until_needlematch 'grub-linux-line', 'down', 15;
+        send_key 'end';
+        send_key 'spc';
+        type_string '3';
+        wait_still_screen(2);
```

Comment:
From experience, especially on aarch64 and power, I think the wait_still_screen is a sensible defensive technique to the problems we often face with the keyboard buffer in qemu

Yes, we know the latest qemu upstream versions address many of those concerns, but we don't have the openQA enablement to make use of the new virt-input magic yet, so I'd rather advocate caution in cases like this and use something similar to what @dgutu-suse proposes